velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -172,6 +179,184 @@ class S3ReadFile final : public ReadFile {
   int64_t length_ = -1;
 };
 
+// AWS C++ SDK allows streaming writes via the MultiPart upload API.
+// Multipart upload allows you to upload a single object as a set of parts.
+// Each part is a contiguous portion of the object's data.
+// You can upload these object parts independently and in any order.
+/// After all parts of your object are uploaded, Amazon S3 assembles these parts
+/// and creates the object.
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html
+// S3WriteFile uses the Apache Arrow implementation as a reference.
+// https://github.com/apache/arrow/blob/main/cpp/src/arrow/filesystem/s3fs.cc
+// S3WriteFile is not thread-safe.
+// UploadPart is currently synchronous during flush.
+// TODO: Evaluate and add option for asynchronous part uploads.
+class S3WriteFile final : public WriteFile {
+ public:
+  explicit S3WriteFile(const std::string& path, Aws::S3::S3Client* client)
+      : client_(client) {
+    bucketAndKeyFromS3Path(path, bucket_, key_);
```

Comment:
nit: Should it be named setBucketAndKeyFromS3Path?

Reply:
Fixed here https://github.com/facebookincubator/velox/pull/6528

Reply:
I named it `getBucketAndKeyFromS3Path` since the function is inferring those values.